### ISSUE TITLE
Smaller package for linking to system openblas

### ISSUE THREAD

**Issue by bluss** [cmr/openblas-provider#9]
This is kind of a minor issue, but the openblas-provider package is a 10MB download and 100MB+ on disk, which is a lot if you just want the linking directive.

**Comment by IvanUkhov** [comment]
Yeah, I know. OpenBLAS isn’t that bad actually; Netlib required additional [stripping](https://github.com/cmr/netlib-provider/blob/master/Cargo.toml#L14) so that it would fit into crates.io’s limits. Anyway, any suggestions would be very much appreciated.

**Comment by IvanUkhov** [comment]
I guess one could just drop `openblas-provider` and do the linking manually in the build script:

```rust
fn main() {
    println!("cargo:rustc-link-lib=dylib=openblas");
}
```

or somewhere in the source code:

```rust
#[link(name = "openblas")]
extern {}
```

**Comment by IvanUkhov** [comment]
Alternatively, we could clone the OpenBLAS repository right from the build script. Since Cargo needs an Internet connection anyway, I guess it’s safe to assume that this connection is available at build time. And Git is kind of a must in the Rust community, so it’s also safe to assume that the git command is available.

**Comment by bluss** [comment]
Hm I think that's not so good, cargo should work without a connection (with a package that was already fetched). Maybe there can simply be another provider crate for the link line. Since cargo's crate feature flags don't really support the alternatives system, I don't think another provider crate has to need adjustment to add it everywhere.
